src/Surfnet/StepupMiddleware/ApiBundle/Identity/Projector/SecondFactorProjector.php

```diff
@@ -234,7 +235,9 @@ public function applySecondFactorVettedEvent(SecondFactorVettedEvent $event)
         $vetted->secondFactorIdentifier = $event->secondFactorIdentifier->getValue();
 
         $this->vettedRepository->save($vetted);
-        $this->verifiedRepository->remove($verified);
+        if ($verified && !$event->vettingType instanceof MoveTokenVettingType) {
```

Comment:
Is it not better to just always remove a token if it exists, no matter what the type is? The state should already be validated in the aggregate, so it's safe to assume you can always remove the token.